lib/fabrik/actions/js/BaseAction.js

```diff
@@ -0,0 +1,23 @@
+'use strict';
+const Promise = require('bluebird');
+
+class BaseAction {
+  static executePreCreate() {
+    return Promise.resolve(0);
+  }
+  static executePostCreate() {
+    return Promise.resolve(0);
+  }
+  static executePreDelete() {
+    return Promise.resolve(0);
+  }
+  static executePostDelete() {
+    return Promise.resolve(0);
+  }
+  static executePreUpdate() {
+    return Promise.resolve(0);
+  }
```

Comment:
any reason to leave out bind/unbind & post update? If there are valid reasons then it could be worth mentioning the same in comments section for this class as one could then know why hook arent defined. If the reason is that they are not need for now, even thats a valid comment.

Reply:
Added PreCreate, PreUpdate, PreDelete PreBind and PreUnbind.

Reply:
Ok